flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/AbstractHandler.java

```diff
@@ -123,7 +124,7 @@ protected void respondAsLeader(ChannelHandlerContext ctx, RoutedRequest routedRe
 			} else {
 				try {
 					ByteBufInputStream in = new ByteBufInputStream(msgContent);
-					request = MAPPER.readValue(in, untypedResponseMessageHeaders.getRequestClass());
+					request = MAPPER.readValue((InputStream) in, untypedResponseMessageHeaders.getRequestClass());
```

Comment:
you can solve this in a nicer way by changing the line above to `InputStream in = new ByteBufInputStream(msgContent);`